geonode/geoserver/views.py

```diff
@@ -365,10 +365,18 @@ def strip_prefix(path, prefix):
             if downstream_path == 'rest/styles':
                 affected_layers = style_update(request, url)
 
-    response, content = http.request(
-        url, request.method,
-        body=request.body or None,
-        headers=headers)
+    if request.method == 'POST':
```

Comment:
This can be simplified and instead of having an `if/elif` statements we could just use something like: https://github.com/boundlessgeo/geonode/blob/exchange/1.5.x/geonode/security/models.py#L89